contracts/JumpBall.cdc

```diff
@@ -48,109 +55,178 @@ access(all) contract JumpBall {
+            for key in keys {
+                let nft <- self.nfts.remove(key: key) ?? panic("NFT not found.")
+                let previousOwner = self.ownership.remove(key: key)!
+                winnerCollection.deposit(token: <-nft)
+                emit NFTAwarded(gameID: self.id, nftID: key, previousOwner: previousOwner, winner: winner)
+            }
+        }
+    }
+
+    // Admin resource for game management
+    access(all) resource Admin {
+        access(all) fun determineWinner(gameID: UInt64, winnerCap: Capability<&{NonFungibleToken.Collection}>, stats: {UInt64: UInt64}) {
+            let game = JumpBall.games[gameID] ?? panic("Game does not exist.")
+
+            let creatorTotal = self.calculateTotal(game: game, stats: stats, player: game.creator)
+            let opponentTotal = self.calculateTotal(game: game, stats: stats, player: game.opponent)
+
+            if creatorTotal > opponentTotal {
+                self.awardWinner(game: game, winner: game.creator, winnerCap: winnerCap)
+            } else if opponentTotal > creatorTotal {
+                self.awardWinner(game: game, winner: game.opponent, winnerCap: winnerCap)
+            } else {
+                self.returnAllNFTs(game: game)
+            }
+        }
+
+        access(self) fun calculateTotal(game: &Game, address: Address, stats: {UInt64: UInt64}): UInt64 {
+            return game.nfts.keys.reduce(0, fun(acc: UInt64, key: UInt64): UInt64 {
+                let owner = game.ownership[key] ?? panic("Owner not found.")
+                if owner == address {
+                    return acc + (stats[key] ?? 0)
+                }
+                return acc
+            })
+        }
+
+        access(self) fun awardWinner(game: &Game, winner: Address, winnerCap: Capability<&{NonFungibleToken.Collection}>) {
+            emit WinnerDetermined(gameID: game.id, winner: winner)
+            game.transferAllToWinner(winner: winner, winnerCap: winnerCap)
         }
     }
 
     // Game registry
     access(self) var nextGameID: UInt64
     access(all) var games: @{UInt64: Game}
-
-    // Mapping of users to their associated gameIDs
     access(all) var userGames: {Address: [UInt64]}
 
     init() {
-        self.nextGameID = 1
-        self.games <- {}
-        self.userGames <- {}
+        JumpBall.nextGameID = 1
+        JumpBall.games <- {}
+        JumpBall.userGames <- {}
     }
 
     // Create a new game
-    access(all) fun createGame(creator: Address, opponent: Address): UInt64 {
-        let gameID = self.nextGameID
-        self.games[gameID] <- create Game(id: gameID, creator: creator, opponent: opponent)
-        self.nextGameID = self.nextGameID + 1
-
-        // Update userGames mapping
-        self.addGameForUser(creator, gameID)
-        self.addGameForUser(opponent, gameID)
-
-        emit GameCreated(gameID: gameID, creator: creator)
+    access(all) fun createGame(creator: Address, startTime: UFix64, gameDuration: UFix64): UInt64 {
```

Comment:
should validation be added for the address parameter, or is the intention that anyone can create a game with any address?

should a player resource be defined an the ability to create a game be scoped to it, so that games can be tied to each player resource's uuid?